src/GoogleSignin.android.js

```diff
@@ -83,100 +80,64 @@ class GoogleSignin {
   }
 
   currentUserAsync() {
-    return new Promise((resolve, reject) => {
-      const sucessCb = DeviceEventEmitter.addListener('RNGoogleSignInSilentSuccess', (user) => {
-        this._user = {...user};
-
-        RNGoogleSignin.getAccessToken(user).then((token) => {
-          this._user = {
-            ...user,
-            accessToken: token
-          };
-          this._removeListeners(sucessCb, errorCb);
-          resolve(this._user);
-        })
-        .catch(err => {
-          this._removeListeners(sucessCb, errorCb);
-          resolve(this._user);
-        });
+    return RNGoogleSignin.currentUserAsync()
+      .then(user => {
+        this._user = { ...user };
+        return user;
+      })
+      .catch(error => {
+        // The user has never signed in before with the given scopes, or they have since signed out.
+        if (error.code === '-4') {
```

Comment:
`'-4'` in iOS means that the user has cancelled the sign in process. On Android the same code is `12501`.

If I remember correctly from iOS side you cannot return integers or numbers as a first parameter of `RCTPromiseRejectBlock` that why I had to do string comparison. We definitely need more robust way of handling errors.